spec/features/incoming_mail_handler_spec.rb

```diff
@@ -0,0 +1,106 @@
+      'email'      => mail_msg.from[0],
+      'sender'     => nil,
+      'text'       => mail_msg.text_part.body.encoded,
+      'html'       => mail_msg.html_part.body.encoded,
+      'raw_msg'    => mail_msg.to_s,
+      'headers'    => headers,
+    }
+    [ { 'event' => 'inbound', 'msg' => msg } ]
+  end
+
+  it "should drop non-app email on the floor" do
+    # this raw json string here as much for example documentation as for test.
+    mandrill_event = <<-END.gsub(/^ {6}/, '')
+      [
+         {
+            "event" : "inbound",
+            "msg" : {
+               "subject" : "test via example server",
+               "template" : null,
+               "tags" : [],
+               "spf" : {
+                  "detail" : "",
+                  "result" : "none"
+               },
+               "from_email" : "someone@example.com",
+               "dkim" : {
+                  "valid" : false,
+                  "signed" : false
+               },
+               "email" : "tester@c2.18f.gov",
+               "sender" : null,
+               "text" : "hello world",
+               "raw_msg" : "Received: from example.com (unknown [93.184.216.34])\\n\\tby ip-xx-xx-xx (Postfix) with ESMTPS id 3EB7C2C06B9\\n\\tfor <tester@c2.18f.gov>; Wed, 23 Sep 2015 21:22:46 +0000 (UTC)\\nReceived: by example.com (Postfix, from userid 500)\\n\\tid 67DDF3E841; Wed, 23 Sep 2015 16:22:45 -0500 (CDT)\\nTo: tester@c2.18f.gov\\nSubject: test via example server\\nMessage-Id: <20150923212245.67DDF3E841@example.com>\\nDate: Wed, 23 Sep 2015 16:22:45 -0500 (CDT)\\nFrom: someone@example.com\\n\\nhello world",
+               "headers" : {
+                  "Received" : [
+                     "from example.com (unknown [93.184.216.34]) by ip-xx-xx-xx (Postfix) with ESMTPS id 3EB7C2C06B9 for <tester@c2.18f.gov>; Wed, 23 Sep 2015 21:22:46 +0000 (UTC)",
+                     "by example.com (Postfix, from userid 500) id 67DDF3E841; Wed, 23 Sep 2015 16:22:45 -0500 (CDT)"
+                  ],
+                  "From" : "someone@example.com",
+                  "Message-Id" : "<20150923212245.67DDF3E841@peknet.com>",
+                  "To" : "tester@c2.18f.gov",
+                  "Subject" : "test via example server",
+                  "Date" : "Wed, 23 Sep 2015 16:22:45 -0500 (CDT)"
+               },
+               "to" : [
+                  [
+                     "tester@c2.18f.gov",
+                     null
+                  ]
+               ],
+               "spam_report" : {
+                  "matched_rules" : [
+                     {
+                        "score" : 1.3,
+                        "name" : "RDNS_NONE",
+                        "description" : "Delivered to internal network by a host with no rDNS"
+                     }
+                  ],
+                  "score" : 1.3
+               }
+            },
+            "ts" : 1443043366
+         }
+      ]
+    END
```

Comment:
Minor: I'd put this in `spec/fixtures/mandrill_incoming.json`, just to make the spec file easier to read.